### PR TITLE
[android][image-picker] fix getting filename and filesize from cropped images

### LINKED ISSUE #27554: [expo-image-picker] Cropped image returns null for fileName and filesize
### Minimal reproducible example

https://github.com/AounNaqvi173/image-picker-example

### What platform(s) does this occur on?

Android

### Did you reproduce this issue in a development build?

Yes

### Summary

expo-image-picker returns null value against fileName and filesize for the edited image. When the allowsEditing is false it works as expected and returns the proper value, I have tested it on Android both a real device and the simulator. See the video for quick demo

https://github.com/expo/expo/assets/75715090/95d96ac0-a200-4a65-b797-71279c1f5619

### Environment

 expo-env-info 1.2.0 environment info:
    System:
      OS: Windows 10 10.0.19045
    Binaries:
      Node: 20.11.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.21 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 9.8.0 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: AI-231.9392.1.2311.11076708
    npmPackages:
      expo: ~50.0.11 => 50.0.11
      react: 18.2.0 => 18.2.0
      react-native: 0.73.4 => 0.73.4
    Expo Workflow: managed

### Expo Doctor Diagnostics

 Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check npm/ yarn versions
✔ Check for issues with metro config
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check that native modules do not use incompatible support packages
✔ Check for legacy global CLI installed locally
✔ Check that native modules use compatible support package versions for installed Expo SDK
✔ Check that packages match versions required by installed Expo SDK

Didn't find any issues with the project!

### PR DESCRIPTION
**Pull request by fobos531**
# Why

Fixes https://github.com/expo/expo/issues/27554

# How

It seems like the current method for fetching filename and file size via contentResolver method results in null values. However, it seems like accessing the File object directly that is acquired via outputFile produces the desired result both with cropped images and original images.

# Test Plan

Tested on Pixel 8 Pro simulator with API 33. Untested on other API levels, but I believe this method should work on other API levels as well. Please let me know if that is not the case.

Note the filename and filesize values for both images with allowsEditing: false and allowsEditing: true

Before

https://github.com/expo/expo/assets/19533289/ca22ab26-1090-4198-9584-268903148d34

After

https://github.com/expo/expo/assets/19533289/0182fc8b-4b6a-49f2-9ed1-67dfb3e37b08

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [X] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/MediaHandler.kt [hunk 1]
```diff
@@ -57,15 +57,14 @@ internal class MediaHandler(
...
 
     return ImagePickerAsset(
       type = MediaType.IMAGE,
       uri = Uri.fromFile(outputFile).toString(),
       width = exportedImage.width,
       height = exportedImage.height,
-      fileName = fileData?.fileName,
-      fileSize = fileData?.fileSize,
+      fileName = outputFile.name,
+      fileSize = outputFile.length(),
```
Comment (alanjhughes) [inline packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/MediaHandler.kt:67]:
```suggestion
      fileName = fileData?.fileName ?: outputFile.name,
      fileSize = fileData?.fileSize ?: outputFile.length(),
```

File: packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/MediaHandler.kt [hunk 2]
```diff
@@ -57,15 +57,14 @@ internal class MediaHandler(
     val exif = options.exif.takeIf { it }
       ?.let { exportedImage.exif(context.contentResolver) }
 
-    val fileData = getAdditionalFileData(sourceUri)
```
Comment (alanjhughes) [inline packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/MediaHandler.kt:60]: Restore this

**Review by alanjhughes — CHANGES_REQUESTED** [review]
Hi @fobos531 - This solution is not correct. What you are returning for the filename is actually just a UUID appended to the file extension that we create when we create the output file, not the actual name of the file. 

The cause of the problem is the cropped image does not exist in the content resolver so the solution is to fallback to using the file as you have as this cropped image will only exist where we wrote it too.

**Comment by fobos531** [comment]
Hello @alanjhughes 

That makes sense! Feedback applied.

**Review by alanjhughes — APPROVED** [review]
Great! Thanks for the fix @fobos531

**Review by lukmccall — APPROVED** [review]
Looks good! Thanks for your contribution🥇

**Pull request merged**